packages/cashier/src/Containers/__tests__/cashier.spec.js

```diff
@@ -0,0 +1,211 @@
+    const original_module = jest.requireActual('@deriv/shared');
+
+    return {
+        ...original_module,
+        isMobile: jest.fn(() => false),
+        WS: {
+            wait: (...payload) => {
+                return Promise.resolve([...payload]);
+            },
+        },
+    };
+});
+
+jest.mock('Components/account-prompt-dialog', () => jest.fn(() => 'mockedAccountPromptDialog'));
+jest.mock('Components/error-dialog', () => jest.fn(() => 'mockedErrorDialog'));
+
+describe('<Cashier />', () => {
+    const props = {
+        is_account_setting_loaded: true,
+        is_account_transfer_visible: true,
+        is_logged_in: true,
+        is_payment_agent_transfer_visible: true,
+        is_payment_agent_visible: true,
+        is_p2p_enabled: true,
+        is_onramp_tab_visible: true,
+        is_visible: true,
+        location: {
+            pathname: '/cashier/deposit',
+        },
+        routes: [
+            {
+                path: '/cashier/deposit',
+                component: jest.fn(() => 'mockedDeposit'),
+                getTitle: () => 'Deposit',
+                icon_component: 'IcCashierAdd',
+                default: true,
+            },
+            {
+                path: '/cashier/withdrawal',
+                component: jest.fn(() => 'mockedWithdrawal'),
+                getTitle: () => 'Withdrawal',
+                icon_component: 'IcCashierMinus',
+            },
+            {
+                path: '/cashier/account-transfer',
+                component: jest.fn(() => 'mockedAccountTransfer'),
+                getTitle: () => 'Transfer',
+                icon_component: 'IcAccountTransfer',
+            },
+            {
+                path: '/cashier/payment-agent',
+                component: jest.fn(() => 'mockedPaymentAgent'),
+                getTitle: () => 'Payment agents',
+                icon_component: 'IcPaymentAgent',
+            },
+            {
+                path: '/cashier/payment-agent-transfer',
+                component: jest.fn(() => 'mockedPaymentAgentTransfer'),
+                getTitle: () => 'Transfer to client',
+                icon_component: 'IcAccountTransfer',
+            },
+            {
+                path: '/cashier/p2p',
+                component: jest.fn(() => 'mockedP2P'),
+                getTitle: () => 'Deriv P2P',
+                icon_component: 'IcDp2p',
+            },
+            {
+                path: '/cashier/on-ramp',
+                component: jest.fn(() => 'mockedOnRamp'),
+                getTitle: () => 'Fiat onramp',
+                icon_component: 'IcCashierOnRamp',
+            },
+        ],
+        routeBackInApp: jest.fn(),
+        onMount: jest.fn(),
+        setAccountSwitchListener: jest.fn(),
+        toggleCashier: jest.fn(),
+    };
+
+    it('should show the loading component if client_tnc_status is not yet loaded or not yet logged in', () => {
+        const history = createBrowserHistory();
```

Comment:
@farrah-deriv Can we remove these `const history = createBrowserHistory();` from each individual test, and define it at the top?

Reply:
I believe `createBrowserHistory` doesn't get reset between tests so i think it's better to declare it on each test.

Reply:
It doesn't but in the 3 beginning tests, it doesn't even change. Only in the last 3 ones, it's changing. It was a suggestion though.